simplq/src/store/asyncActions/joinQueue.js

```diff
@@ -21,7 +21,9 @@ const useJoinQueue = () => {
         RequestFactory.createToken(name, contactNumber, notifiable, queueId)
       );
       const response = await authedRequest;
-      history.push(`/token/${response.tokenId}`);
+      if (response) {
+        history.push(`/token/${response.tokenId}`);
+      }
       return response;
```

Comment:
I'm not sure if this condition is required, redux seems to implicitly handle cases where this request fails.

Reply:
I am also not sure, I'd vote to remove the condition and fix later if there is an issue.

Reply:
[createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk#return-value) always resolves, and dispatches `pending` action first, and then `fulfilled` or `rejected` action. `try catch` is needed only if you want to handle rejected promise here.

Reply:
We'll need to add try-catches at a few places including this one. We used to display API failures using [handleApiErrors](https://github.com/SimplQ/simplQ-frontend/blob/7e8c77088cf18ea66a0de918c127edbc49281ace/simplq/src/api/axios.js#L8), but this doesn't work with our redux setup. So we'll either have to use this function in [makeAuthedRequest](https://github.com/SimplQ/simplQ-frontend/blob/7e8c77088cf18ea66a0de918c127edbc49281ace/simplq/src/api/auth.js#L23) or add try-catches and extraReducers to AppSlice wherever required.

Reply:
The actionStatusSlice way is a better approach.